sdk/web-pubsub/web-pubsub-client/test/client.spec.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "@azure/test-utils";
+import { WebPubSubClientOptions } from "../src/models";
+import { WebPubSubJsonProtocol } from "../src/protocols";
+import { WebPubSubClient } from "../src/webPubSubClient";
+import { DefaultWebPubSubClientCredential } from "../src/webPubSubClientCredential";
+
+describe("WebPubSubClient", function () {
+  describe("Construct a new client and options", () => {
+    it("takes a client access url", () => {
+      assert.doesNotThrow(() => {
+        new WebPubSubClient("wss://service.com");
+      });
+    });
+
+    it("take client access url as func", () => {
+      assert.doesNotThrow(() => {
+        new WebPubSubClient(new DefaultWebPubSubClientCredential(async (_) => "wss://service.com"));
```

Comment:
this feels more like an endpoint uri than a credential -- also not clear what the scenario is for the provider since a client could lazily be instantiated

Reply:
It's a uri but contains token in query string. The function is called every time the client starts a new WebSocket connection (The WebSocket connection may drop and reconnect).